bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtocol.java

```diff
@@ -445,7 +443,7 @@ void recycle() {
     /**
      * A request that reads data.
      */
-    class ReadResponse extends Response {
+    class ReadResponse extends Response implements ReferenceCounted {
```

Comment:
why do we need to implement ReferenceCounted ?
can you add a comment in the JavaDoc ?

Reply:
https://github.com/apache/bookkeeper/issues/3527 will give more context.